Assets/MixedRealityToolkit.SDK/Features/Input/Handlers/EyeTrackingTarget.cs

```diff
@@ -3,69 +3,186 @@
+        private float dwellTimeInSec = 0.8f;
+
+        [Tooltip("Event is triggered when the user starts to look at the target.")]        
+        public UnityEvent OnLookAtStart = null;
 
         [Tooltip("Event is triggered when the user continues to look at the target.")]
-        [SerializeField]
-        private UnityEvent WhileLookingAtTarget = null;
+        public UnityEvent WhileLookingAtTarget = null;
 
         [Tooltip("Event to be triggered when the user is looking away from the target.")]
-        [SerializeField]
-        private UnityEvent OnLookAway = null;
+        public UnityEvent OnLookAway = null;
 
         [Tooltip("Event to be triggered when the target has been looked at for a given predefined duration.")]
-        [SerializeField]
-        private UnityEvent OnDwell = null;
+        public UnityEvent OnDwell = null;
 
         [Tooltip("Event to be triggered when the looked at target is selected.")]
-        [SerializeField]
-        private UnityEvent OnSelected = null;
+        public UnityEvent OnSelected = null;
+        
+        public bool IsLookedAt { get; private set; }
+
+        private bool isDwelledOn = false;
+        /// <summary>
+        /// Returns true if the user has been looking at the target for a certain amount of time specified by dwellTimeInSec.
+        /// </summary>
+        public bool IsDwelledOn
+        {
+            get { return isDwelledOn; }
+        }
+
+        private IMixedRealityInputSystem inputSystem = null;
+
+        /// <summary>
+        /// The active instance of the input system.
+        /// </summary>
+        private IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
+        
+        private DateTime lookAtStartTime;
+        private float EyeTrackerFramerate = 30; // In Hz -> This means that every 1000ms/30 = 33.33ms a new sample should arrive. 
+        private float EyeTrackingTimeoutInMilliseconds = 100; // To account for blinks
+
+        private static DateTime lastTimeStamp = DateTime.MinValue;
+        public static GameObject LookedAtTarget { get;  private set; }
+        public static EyeTrackingTarget LookedAtEyeTarget { get; private set; }
+        public static Vector3 LookedAtPoint { get; private set; }
+
+        #region Focus handling
+        private void Start()
+        {
+            IsLookedAt = false;
+            LookedAtTarget = null;
+            LookedAtEyeTarget = null;
+        }
+        private void Update()
+        {
+            // Try to manually poll the eye tracking data
+            if ((InputSystem?.EyeGazeProvider != null) && (InputSystem?.EyeGazeProvider?.UseEyeTracking == true) && (InputSystem?.EyeGazeProvider?.IsEyeGazeValid == true))
```

Comment:
```suggestion
            if ((InputSystem?.EyeGazeProvider != null) && (InputSystem?.EyeGazeProvider?.UseEyeTracking) && (InputSystem?.EyeGazeProvider?.IsEyeGazeValid))
```

Reply:
If I do this, I get the following error:
![image](https://user-images.githubusercontent.com/32180789/58902211-85985180-86b7-11e9-97cc-f1e7e2dc43fc.png)

Reply:
Oh noooooooooooo yeah I was wrong.